src/Sulu/Bundle/SnippetBundle/Twig/SnippetTypeTwigExtension.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\Twig;
+
+use Sulu\Bundle\SnippetBundle\Snippet\SnippetRepository;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+
+/**
+ * Provides global snippets by type.
+ */
+class SnippetTypeTwigExtension extends \Twig_Extension
+{
+    /**
+     * @var SnippetRepository
+     */
+    private $snippetRepository;
+
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @param SnippetRepository $snippetRepository
+     * @param RequestAnalyzerInterface $requestAnalyzer
+     */
+    public function __construct(SnippetRepository $snippetRepository, RequestAnalyzerInterface $requestAnalyzer)
+    {
+        $this->snippetRepository = $snippetRepository;
+        $this->requestAnalyzer = $requestAnalyzer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return [
+            new \Twig_SimpleFunction('sulu_snippet_load_by_type', [$this, 'loadByType']),
+        ];
+    }
+
+    /**
+     * Load global snippets by type.
+     *
+     * @param string type
+     * @param string $locale
+     *
+     * @return array
+     */
+    public function loadByType($type, $locale = null)
+    {
+        // ensure locale
+        if (null === $locale) {
+            $locale = $this->requestAnalyzer->getCurrentLocalization()->getLocale();
+        }
+
+        // practical helper
+        $helper = [];
+
+        // get all snippets by their type
+        if ($snippets = $this->snippetRepository->getSnippets($locale, $type)) {
```

Comment:
The number of snippets of a selection is limited to approx. 200, because the QueryParam with the selected ids in the url becomes too long. Can snippet areas be loaded without explicit assignment from a content manager?

Reply:
A area only need to be assign once in the webspace settings. I think this can also be done with fixtures. But if you have 200 of them I maybe missunderstand your usecase. Loading 200 snippets once will also take a very long time to be loaded didn't you have some performance problems there? Thought you have only some snippets with blocks to translate the app. Are the snippets used to extend the app with new content?